Sources/MapboxNavigation/NavigationLocationProvider.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+import CoreLocation
+import MapboxCoreNavigation
+import MapboxMaps
+
+/**
+ An object that notifies a map view when the user’s location changes, minimizing the noise that normally accompanies location updates from a `CLLocationManager` object.
+ 
+ If your application displays a `MapView` before starting turn-by-turn navigation, call `LocationManager.overrideLocationProvider(with:)` to override default location provider so that the map view always shows the location snapped to the road network. For example, use this class to show the user’s current location as they wander around town.
+ 
+ This class depends on `NavigationLocationManager` to detect the user’s location as it changes.
+ */
+open class NavigationLocationProvider: NSObject, LocationProvider, CLLocationManagerDelegate {
+    /**
+     The location provider's location manager, which detects the user’s location as it changes.
+     */
+    public var locationManager: NavigationLocationManager
```

Comment:
NavigationLocationProvider contains NavigationLocationManager, similar to the way PassiveLocationManager contains PassiveLocationDataSource. But it’s a bit confusing that NavigationLocationProvider and PassiveLocationManager are both LocationProviders, considering their names. Should we rename PassiveLocationManager to PassiveLocationProvider, and if so, should we rename PassiveLocationDataSource to PassiveLocationManager for consistency with the other NavigationLocationManager classes?

Reply:
Personally,  because `PassiveLocationManager ` has been used for a long time and works pretty fine so far. I think maybe that the `PassiveLocationManager` should not be subclass of `NavigationLocationProvider `. Because `NavigationLocationProvider `  should be responsible to provide location update during turn-by-turn navigation for camera and puck instead of snapped location (provided by default `AppleLocationProvider`). They should at be the same level, one for free-drive, and one for turn-by-turn active guidance navigation.

Reply:
Sure, PassiveLocationManager doesn’t have to be a subclass of NavigationLocationProvider; the main benefit was the ability to remove some redundant implementations of methods like `requestWhenInUseAuthorization()`, but they aren’t very large implementations anyways. Essentially my feedback is about naming: PassiveLocationManager and NavigationLocationProvider could be siblings and it would still be confusing that only one is called a “provider” even though both are LocationProviders and neither is a CLLocationManager.

Reply:
I see. Thanks for the suggestion! I'll rename the `PassiveLocationManager ` to `PassiveLocationProvider ` .